orbit/cmd/orbit/orbit.go

```diff
@@ -223,6 +224,34 @@ func main() {
 			}
 		}
 
+		// if neither are set, this might be an agent deployed via MDM, try to read
+		// both configs from a configuration profile
+		if c.String("fleet-url") == "" && c.String("enroll-secret") == "" {
+			config, err := profiles.GetFleetdConfig()
+			switch {
+			// handle these errors separately as debug messages to not raise false
+			// alarms when users look into the orbit logs, it's perfectly normal to
+			// not have a configuration profile, or to get into this situation in
+			// operating systems that don't have profile support.
+			case errors.Is(err, profiles.ErrNotImplemented):
+			case errors.Is(err, profiles.ErrNotFound):
+				log.Debug().Msgf("reading configuration profile: %e", err)
+			case err != nil:
+				// TODO: should we retun here? that will make orbit exit
+				log.Error().Err(err).Msg("reading configuration profile")
+			case config.EnrollSecret == "" || config.FleetURL == "":
+				log.Debug().Msg("enroll secret or fleet url are empty in configuration profile, not setting either")
```

Comment:
am I abusing the `switch` here? happy to move this into an `if` in the `default` clause.